core/dbt/adapters/base/connections.py

```diff
@@ -5,23 +5,26 @@
 
 import dbt.exceptions
 import dbt.flags
-from dbt.api import APIObject
 from dbt.contracts.connection import Connection
+from dbt.contracts.util import Replaceable
 from dbt.logger import GLOBAL_LOGGER as logger
 from dbt.utils import translate_aliases
 
+from hologram.helpers import ExtensibleJsonSchemaMixin
 
-class Credentials(APIObject):
-    """Common base class for credentials. This is not valid to instantiate"""
-    SCHEMA = NotImplemented
-    # map credential aliases to their canonical names.
-    ALIASES = {}
+from dataclasses import dataclass, field
+from typing import Any, ClassVar, Dict, Tuple
 
-    def __init__(self, **kwargs):
-        renamed = self.translate_aliases(kwargs)
-        super().__init__(**renamed)
 
-    @property
+@dataclass
+class Credentials(
+    ExtensibleJsonSchemaMixin,
+    Replaceable,
+    metaclass=abc.ABCMeta
+):
+    _ALIASES: ClassVar[Dict[str, str]] = field(default={}, init=False)
```

Comment:
don't you want `field(default_factory=dict, ...)` here

Reply:
Nope! This is a ClassVar, as in a static class-level attribute, so `default` is appropriate. It's not called on instantiation by the dataclass layer (hence `init=False`). Though I do think I could maybe do this instead?:
```
_ALIASES: ClassVar[Dict[str, str]] = {}
```

It seems like dataclasses correctly figure out when they're class vars that they don't get an `__Init__` entry

Reply:
Ok, I was pretty wrong about doing the above instead! When that happens, everything looks great until you call `dataclasses.replace(credentials, ...)` - then python throws a TypeError complaining that it got an unexpected keyword argument `_ALIASES`. That seems like a bug to me.